src/main/java/com/google/api/codegen/discovery/transformer/js/JSSampleMethodToViewTransformer.java

```diff
@@ -0,0 +1,159 @@
+ */
+package com.google.api.codegen.discovery.transformer.js;
+
+import com.google.api.codegen.discovery.config.FieldInfo;
+import com.google.api.codegen.discovery.config.MethodInfo;
+import com.google.api.codegen.discovery.config.SampleConfig;
+import com.google.api.codegen.discovery.transformer.SampleMethodToViewTransformer;
+import com.google.api.codegen.discovery.transformer.SampleNamer;
+import com.google.api.codegen.discovery.transformer.SampleTransformerContext;
+import com.google.api.codegen.discovery.transformer.SampleTypeTable;
+import com.google.api.codegen.discovery.transformer.nodejs.NodeJSSampleNamer;
+import com.google.api.codegen.discovery.transformer.nodejs.NodeJSSampleTypeNameConverter;
+import com.google.api.codegen.discovery.viewmodel.SampleAuthView;
+import com.google.api.codegen.discovery.viewmodel.SampleFieldView;
+import com.google.api.codegen.discovery.viewmodel.SamplePageStreamingView;
+import com.google.api.codegen.discovery.viewmodel.SampleView;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.SymbolTable;
+import com.google.api.codegen.util.nodejs.NodeJSNameFormatter;
+import com.google.api.codegen.util.nodejs.NodeJSTypeTable;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.protobuf.Field.Cardinality;
+import com.google.protobuf.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JSSampleMethodToViewTransformer implements SampleMethodToViewTransformer {
+
+  private static final String TEMPLATE_FILENAME = "js/sample.snip";
+
+  public JSSampleMethodToViewTransformer() {}
+
+  @Override
+  public ViewModel transform(Method method, SampleConfig sampleConfig) {
+    // Since the conventions between Node.js and Javascript are essentially
+    // identical, we reuse existing Node.js components.
+    SampleTypeTable sampleTypeTable =
+        new SampleTypeTable(new NodeJSTypeTable(""), new NodeJSSampleTypeNameConverter());
+    NodeJSSampleNamer nodeJSSampleNamer = new NodeJSSampleNamer();
+    SampleTransformerContext context =
+        SampleTransformerContext.create(
+            sampleConfig, sampleTypeTable, nodeJSSampleNamer, method.getName());
+    return createSampleView(context);
+  }
+
+  private SampleView createSampleView(SampleTransformerContext context) {
+    SampleConfig config = context.getSampleConfig();
+    MethodInfo methodInfo = config.methods().get(context.getMethodName());
+    SampleNamer namer = context.getSampleNamer();
+    SampleTypeTable typeTable = context.getSampleTypeTable();
+    SymbolTable symbolTable = SymbolTable.fromSeed(NodeJSNameFormatter.RESERVED_IDENTIFIER_SET);
+
+    SampleView.Builder builder = SampleView.newBuilder();
+
+    String requestVarName = symbolTable.getNewSymbol(namer.getRequestVarName());
+
+    if (methodInfo.isPageStreaming()) {
+      builder.pageStreaming(createSamplePageStreamingView(context, symbolTable));
+    }
+
+    List<SampleFieldView> fields = new ArrayList<>();
```

Comment:
Not in this one and some others because either the auth portions create no variables, or they're members of another scope.